eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Gov.hs

```diff
@@ -240,23 +261,42 @@ actionWellFormed ga = failureUnless isWellFormed $ MalformedProposal ga
       ParameterChange _ ppd -> ppuWellFormed ppd
       _ -> True
 
+-- | Called from within `addAction` to check if the proposal being added is the first one
+-- after a set of contiguous epochs (we need to check only the previous epoch to be sure)
+-- that had no active governance proposals to vote on, so that `DRep` expiry can be updated
+-- and the dormant-epochs counter can be reset to zero.
+-- Although this function is called on each proposal, it makes sure to update the `DRep`
+-- expiry only the first time that a new proposal is accepted in an epoch.
+updateDRepExpiry :: GovRuleState era -> GovRuleState era
+updateDRepExpiry grs@(GovRuleState govSnapshots vState) =
```

Comment:
Just checking... it's not possible to do this in `Certs`, where we also update the expiry of Dreps after voting? 
If that was possible, then we wouldn't have to have the VState in the GovState (not saying that the latter is necessarily bad btw).

Reply:
> it's not possible to do this in Certs rule, where we also update the expiry of Dreps after voting?

I don't think so, because we don't have access to the `GovSnapshots` in `CERTS`. We'd have to make some unavailable state available to the target rule either way, if I'm not mistaken.